lib/SampleService/core/validator/builtin.py

```diff
@@ -321,3 +323,67 @@ def _is_num(name, val):
     if val is not None and type(val) != float and type(val) != int:
         raise ValueError(f'Value for {name} parameter is not a number')
     return val
+
+def ontology_has_ancestor(d: Dict[str, Any]) -> Callable[[str, Dict[str, PrimitiveType]], Optional[str]]:
+    '''
+    Build a validation callable that checks that value has valid ontology ancestor term provided
+
+    The 'ontology' parameter is required and must be a string. It is the ontology name.
+
+    The 'ancestor_term' parameter is required and must be a string. It is the ancestor name.
+
+    The 'kbase_endpoint' parameter is required and must be a string. It is kbase endpoint url for getting OntologyAPI service.
+
+    :param d: the configuration map for the callable.
+    :returns: a callable that validates metadata maps.
+    '''
+    _check_unknown_keys(d, {'ontology', 'ancestor_term', 'kbase_endpoint'})
+
+    ontology = d.get('ontology')
+    if not ontology:
+        raise ValueError('ontology is a required paramter')
+    if type(ontology) != str:
+        raise ValueError('ontology must be a string')
+
+    ancestor_term = d.get('ancestor_term')
+    if not ancestor_term:
+        raise ValueError('ancestor_term is a required paramter')
+    if type(ancestor_term) != str:
+        raise ValueError('ancestor_term must be a string')
+
+    endpoint=d.get('kbase_endpoint')
+    if not endpoint:
+        raise ValueError('kbase_endpoint is a required paramter')
+    if type(endpoint) != str:
+        raise ValueError('kbase_endpoint must be a string')
+
+    srv_wiz_url=endpoint.strip('/') + '/service_wizard'
+    oac=None
+    try:
+        oac=OntologyAPI(srv_wiz_url)
+        if oac.status().get('state') != 'OK':
+            raise ValueError('the status of ontology api service is not ok')
+    except:
+        raise ValueError('failed connect to ontology api through kbase_endpoint')
+
+
+    def _get_ontology_ancestors(ontology, val):
+        try:
+            ret=oac.get_ancestors({"id": val, "ns": ontology})
+            return list(map(lambda x: x["term"]["id"], ret["results"]))
+        except ServerError as err:
+            if 'InvalidParamsError' in err.data:
+                raise ValueError(f'ontology {ontology} is not valid')
```

Comment:
This doesn't report ontology_term is not validated. It reports the ontology user tried to do the search in is not available in ontology service. I changed it from `not found` to 'not valid` and maybe a better wording could be used.

Reply:
Ok, so if I understand correctly, `InvalidParamsError` means that the user tried to query an ontology that doesn't exist (which seems like a broad error name for a very specific error).

In that case, since the ontology name is provided at startup, it seems to me it'd be better to make sure the ontology exists at startup and fail early.

Reply:
Same with ancestor term

Reply:
updated in new commit